### PR TITLE
Check get_user_locale() exists before attempting to use it, compatibility fix for WP4.5

### LINKED ISSUE #508: Fatal error on WP4.5 - Call to undefined function EventEspresso\core\services\assets\get_user_locale()
Currently master will give a fatal error on the plugins page when using WP 4.5:

```
Fatal error: Uncaught Error: Call to undefined function EventEspresso\core\services\assets\get_user_locale() in D:\xampp\htdocs\wp45\wp-content\plugins\event-espresso-core-reg\core\services\assets\I18nRegistry.php:237 Stack trace: 
#0 D:\xampp\htdocs\wp45\wp-content\plugins\event-espresso-core-reg\core\services\assets\I18nRegistry.php(181): EventEspresso\core\services\assets\I18nRegistry->getJedLocaleData('event_espresso') 
#1 D:\xampp\htdocs\wp45\wp-content\plugins\event-espresso-core-reg\core\services\assets\I18nRegistry.php(139): EventEspresso\core\services\assets\I18nRegistry->getJedLocaleDataForDomainAndChunk('ee-wp-plugins-p...', 'event_espresso') 
#2 D:\xampp\htdocs\wp45\wp-content\plugins\event-espresso-core-reg\core\services\assets\I18nRegistry.php(98): EventEspresso\core\services\assets\I18nRegistry->queueI18nTranslationsForHandle('ee-wp-plugins-p...')
#3 D:\xampp\htdocs\wp45\wp-includes\plugin.php(235): EventEspresso\core\services\assets\I18nRegistry->queueI18n(Array)
#4 D:\xampp\htdocs\wp45\wp-includes\class.wp-s in D:\xampp\htdocs\wp45\wp-content\plugins\event-espresso-core-reg\core\services\assets\I18nRegistry.php on line 237
```

This is a clean install using WP 4.5.14

### PR DESCRIPTION
**Pull request by Pebblo**
This fixes #508

## Problem this Pull Request solves
EE throws a fatal error on WP4.5 as its using a function that doesn't exist.

## How has this been tested
Activated EE on WP4.5 and confirmed no fatal in the admin.

## Checklist

* [x] I have added a changelog entry for this pull request
* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [x] My code is tested.
* [x] My code follows the Event Espresso code style.
* [x] My code has proper inline documentation.
* [ ] My code accounts for when the site is Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

**Comment by joshfeck** [comment]
I tested this locally on a site with WP 4.5 and this branch resolves the issue for me.

In the future, it'd be best if you can avoid using special characters like `()` because command line git requires remembering to escape characters like those.

File: core/services/assets/I18nRegistry.php [hunk 1]
```diff
@@ -231,10 +231,17 @@ private function getJedLocaleData($domain)
     {
         $translations = get_translations_for_domain($domain);
 
+        // check get_user_locale() exists before using it for compatibility with WP4.5
+        if(is_admin() && function_exists('get_user_locale')) {
+            $lang = get_user_locale();
+        } else {
+            $lang = get_locale();
+        }
```
Comment (nerrad) [inline core/services/assets/I18nRegistry.php]: We already have a helper in `EEH_DTT_Helper` for this.  So you can simply modify the existing conditional below to:
```
'lang' => is_admin() ? EEH_DTT_Helper::get_user_locale() : get_locale()
```

**Comment by Pebblo** [comment]
This works fine for me, @joshfeck do you want to restest now it's using the helper?

**Comment by joshfeck** [comment]
It's throwing a new fatal error with this change:

```Fatal error: Uncaught Error: Class 'EventEspresso\core\services\assets\EEH_DTT_Helper' not found in /srv/www/oldwp/public_html/wp-content/plugins/event-espresso-core/core/services/assets/I18nRegistry.php:237 Stack trace: #0 /srv/www/oldwp/public_html/wp-content/plugins/event-espresso-core/core/services/assets/I18nRegistry.php(181): EventEspresso\core\services\assets\I18nRegistry->getJedLocaleData('event_espresso') #1 /srv/www/oldwp/public_html/wp-content/plugins/event-espresso-core/core/services/assets/I18nRegistry.php(139): EventEspresso\core\services\assets\I18nRegistry->getJedLocaleDataForDomainAndChunk('ee-wp-plugins-p...', 'event_espresso') #2 /srv/www/oldwp/public_html/wp-content/plugins/event-espresso-core/core/services/assets/I18nRegistry.php(98): EventEspresso\core\services\assets\I18nRegistry->queueI18nTranslationsForHandle('ee-wp-plugins-p...') #3 /srv/www/oldwp/public_html/wp-includes/plugin.php(235): EventEspresso\core\services\assets\I18nRegistry->queueI18n(Array) #4 /srv/www/oldwp/public_html/wp-includes/cl in /srv/www/oldwp/public_html/wp-content/plugins/event-espresso-core/core/services/assets/I18nRegistry.php on line 237```

**Comment by nerrad** [comment]
Should probably have @tn3rb comment on the new fatal.  My understanding was that helpers were exposed everywhere for use, but something might have changed recently (that I'm not aware of), in which case the helper will need to be  injected as a dependency i18nRegistry.  Although I don't like the idea of having to inject the entire helper for one method use.

**Comment by mnelson4** [comment]
Just chiming in while watching cloudy with a chance of meatballs...
I think we’re just missing `include ‘EEH_DTT_Helper';` as the class isn’t namespaced

**Comment by nerrad** [comment]
oh dude totally right!

so @Pebblo you just need to implement with `\EEH_DTT_Helper` or in the import block at the top of the file add `use EEH_DTT_Helper`.

**Comment by Pebblo** [comment]
Thanks guys, the above commit fixed this for me.

**Comment by Pebblo** [comment]
No worries :) Like so?

**Comment by Pebblo** [comment]
Fatal fixed, this seems to be working fine for me.

@joshfeck want to do another quick test before its merged?

**Comment by joshfeck** [comment]
Verified the fix.

**Pull request merged**